pyam/iiasa.py

```diff
@@ -23,74 +25,114 @@
  data as provided in the explorer guidelines: {}
 """.replace('\n', '')
 
+# path to local configuration settings
+DEFAULT_IIASA_CREDS = Path('~').expanduser() / '.local' / 'pyam' / 'iiasa.yaml'
+
+
+def set_config(user, password, file=None):
+    """Save username and password for IIASA API connection to file"""
+    file = Path(file) if file is not None else DEFAULT_IIASA_CREDS
+    if not file.parent.exists():
+        file.parent.mkdir(parents=True)
+
+    with open(file, mode='w') as f:
+        logger.info(f'Setting IIASA-connection configuration file: {file}')
+        yaml.dump(dict(username=user, password=password), f)
+
+
+def _get_config(file=None):
+    """Read username and password for IIASA API connection from file"""
+    file = Path(file) if file is not None else DEFAULT_IIASA_CREDS
+    if file.exists():
+        with open(file, 'r') as stream:
+            return yaml.safe_load(stream)
+
 
 def _check_response(r, msg='Trouble with request', error=RuntimeError):
     if not r.ok:
         raise error('{}: {}'.format(msg, str(r.text)))
 
 
 def _get_token(creds, base_url):
-    if creds is None:  # get anonymous auth
+    """Parse credentials and get token from IIASA authentication service"""
+    plaintextcreds = True
+
+    # try reading default config or parse file
+    if creds is None:
+        creds = _get_config()
+        plaintextcreds = False
+    elif isinstance(creds, Path) or isstr(creds):
+        _creds = _get_config(creds)
```

Comment:
nit: maybe rename `creds` to `creds_or_config_path`

Reply:
I don't think that passing a custom path to a creds file will be used much (other than by us for developing), because you only have one user-password anyway... so I'd prefer to keep the API concise.